### PR TITLE
Added internal transitions

### PR DESCRIPTION
**Pull request by HenningNT**
Added internal transitions.

### CODE REVIEW

File: Stateless/StateConfiguration.cs [hunk 1]
```diff
@@ -47,7 +47,44 @@ public StateConfiguration Permit(TTrigger trigger, TState destinationState)
...
                 return InternalPermit(trigger, destinationState, string.Empty);
             }
-
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="trigger"></param>
+            /// <param name="entryAction"></param>
+            /// <returns></returns>
+            public StateConfiguration InternalTransistion(TTrigger trigger, Action<Transition> entryAction)
```
Comment (tlk) [inline Stateless/StateConfiguration.cs]: Maybe rename to InternalTransition without the extra s :)

File: Stateless/StateMachine.cs [hunk 2]
```diff
@@ -208,6 +208,15 @@ void InternalFire(TTrigger trigger, params object[] args)
...
                 newRepresentation.Enter(transition, args);
             }
+            else
+            {
+                var transition = new Transition(source, destination, trigger);
+
+                CurrentRepresentation.InternalAction(transition, args);
+
+//                if (_onTransitioned != null)
+//                    _onTransitioned(transition);
```
Comment (tlk) [inline Stateless/StateMachine.cs]: These comments could be removed, right?

File: Stateless/StateRepresentation.cs [hunk 3]
```diff
@@ -207,6 +238,11 @@ public IEnumerable<TTrigger> PermittedTriggers
                     return result.ToArray();
                 }
             }
+            internal void InternalAction(Transition transition, object[] args)
+            {
+                Enforce.ArgumentNotNull(transition, "transtion");
```
Comment (tlk) [inline Stateless/StateRepresentation.cs]: "transtion" -> "transition"

**Comment by HenningNT** [comment]
I tried committing the fixes with Github Desktop app, but it somehow failed. The app tells me the changes are committed, but the link provided by the app leads to a 404 not found :(

Any ideas?

**Comment by tlk** [comment]
Hi @HenningNT I'm not acquinted with the Github Desktop app, and forgive me if this is silly - but perhaps you need to push the locally committed changes to your github repository? http://stackoverflow.com/questions/2745076/what-are-the-differences-between-git-commit-and-git-push

**Comment by HenningNT** [comment]
Yay!
Thanks, that helped. There's a "Sync" button that does the push.

I should probably just use the Git functionality already present in VS2015...

**Comment by HenningNT** [comment]
What happens now?

**Comment by nblumhardt** [comment]
Nice job! Looks like a great feature.

Just a couple of comments:

 * A few tests laying out how the behaviour should work would be beneficial, especially when hierarchical states are involved
 * Since no transition actually takes place, is the name the right one?

**Comment by HenningNT** [comment]
Thanks!
I'll add some tests to make sure it works 😉

The wikipedia page about uml and state machines refer to these event as internal transitions, so I think the name is ok.

**Comment by nblumhardt** [comment]
Looking good. Ready to go?

**Comment by HenningNT** [comment]
Yes!

**Pull request merged**